### PR TITLE
retro from #87: smoke as executor task, KMP UI principle, retro reformulation

### PR DESCRIPTION
**Pull request by khmelevartem**
## Summary

System-level changes informed by PR #87 retro. No production code touched — only `.claude/` and `docs/`.

**Three threads:**

1. **Smoke is the executor's job, not the reviewer's.** `/work-on-issue` Step 7 — implementer runs relevant smoke blocks before review and records verdict in PR body. Step 8 — implementer decides if new critical functionality belongs in the smoke skill itself (without bloating). Static verdict-existence check stays in code review going forward.

2. **KMP UI cross-platform reuse codified.** `architecture-principles.md` gets an explicit principle: Compose Multiplatform UI in `commonMain` ships to all targets with a Compose entry point. Operational rules: common-first, no per-platform issues for common UI, size:* per feature, mandatory manual smoke on each shipping platform. Confirmed in practice with #7.

3. **Retro triggers shift agent-centric → system-centric** (in `retro.md` + `close-issue.md` Step 8). Pointwise mistakes caught by review = system working, not a retro signal. Focus on systemic gaps (doc-vs-reality) and systemic wins worth encoding.

**Plus smaller pieces:**

- New `close-issue.md` Step 4.2 rule: doc-as-spec consistency — when first real implementation lands for an architectural sketch, the doc must be updated in the same PR.
- New `work-on-issue.md` Step 4 sub-step: verify dependency status before adding it (deprecated / current / alternatives), record verdict in PR body.
- `features/README.md` + `device-list.md`: iOS marked done via KMP UI reuse.

**Not in this PR:** restructuring of `/code-review` into multi-agent form — needs proper architecture (skill + `.claude/agents/` subagent types), tracked as a separate task.

## Test plan

- [ ] Skim each modified file end-to-end for internal consistency
- [ ] Verify retro triggers in `retro.md` and `close-issue.md` Step 8 match each other
- [ ] Verify smoke verdict format in `work-on-issue.md` matches what will be checked by future review (verdict section exists in PR body)
- [ ] Spot-check `architecture-principles.md` KMP UI section reads as a principle, not as a one-off observation

### CODE REVIEW

File: .claude/commands/work-on-issue.md [hunk 1]
```diff
@@ -46,3 +62,36 @@
...
+   ## Smoke verdict
+   - Block 1 (Desktop CLI): 🟢
+   - Block 3 (Android FGS, manual rotation + Stop button): 🟢
+   - Block 4 (native compile macosArm64 + iosSimulatorArm64): 🟢
+   - Manual iOS UI (KMP UI reuse): 🟢 — opened simulator, list rendered, EmptyState shown
+   - Skipped: blocks 2 (no FileServer/CLI changes), block 5 (no manual-only path changed)
+   ```
+   Если что-то 🟡/🔴 — **не отправляй в ревью** до резолва.
+
+8. **расширение smoke под новую критическую функциональность.** Сразу после smoke self-check'а, пока контекст реализации свежий, ответь себе: **реализована ли в этом PR новая критическая функциональность, которую следует включить в `/smoke-test`?** Критерий «критическая» — это:
```
Comment (khmelevartem) [inline .claude/commands/work-on-issue.md]: почему расширение смоука после прогона смоука? давай перед

File: .claude/commands/work-on-issue.md [hunk 2]
```diff
@@ -46,3 +62,36 @@
...
+
+   - точка отказа на старте (entry point, init order, theme, manifest);
+   - happy-path фичи, без которой остальное теряет смысл (send roundtrip, FGS живёт, mDNS публикует/находит);
+   - cross-platform reusable UI, у которого ломка на одной платформе незаметна без её прогона;
+   - новый внешний интерфейс (CLI команда, новый HTTP endpoint), который другие компоненты будут использовать.
+
+   Если да — расширь `.claude/skills/smoke-test/SKILL.md` соответствующим блоком (или дополни существующий), в этом же PR. Если расширение крупное — отдельным issue типа INFRA с пометкой «следствие #<этой задачи>».
+
+   **Не раздувай smoke ради полноты.** Smoke — это «за 1-3 минуты увидеть, что ничего фундаментально не сломано», не полная регресс-сюита. Если новая функциональность некритическая (UI косметика, дополнительный edge case, optional path) — её место в unit/integration тестах, не в smoke. Каждый добавленный блок smoke удорожает каждый будущий close-issue.
```
Comment (khmelevartem) [inline .claude/commands/work-on-issue.md]: все, что описано тобой в этом файле - слишком подробно. давай короче, не надо раздувать

File: docs/engineering/architecture-principles.md [hunk 3]
```diff
@@ -68,6 +68,21 @@ When refactoring existing code, the same questions apply in reverse: a layer tha
 - **Presentation layer is built on [Decompose](https://github.com/arkivanov/Decompose).** Components hold state and lifecycle in plain Kotlin; Compose subscribes via `subscribeAsState` and is treated as a thin, replaceable renderer. Conventions and how to write/test components: [presentation-layer.md](presentation-layer.md). Rationale, alternatives considered, and per-platform notes: [adr/adr-presentation-and-navigation.md](adr/adr-presentation-and-navigation.md).
 - **Unidirectional data flow** (state down, events up) is the default. The specific framework — MVI library, Molecule, plain Compose state — is chosen per component as it appears, not declared globally up front.
 
+## Compose Multiplatform UI: common-first, cross-platform reuse
```
Comment (khmelevartem) [inline docs/engineering/architecture-principles.md]: надо поднять перед Named classes over anonymous objects.

также это касается не только UI

также сожми

File: docs/product/features/device-list.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 **Area:** UI
 **Status:** `in progress`
-**GitHub Issues:** [#7](https://github.com/khmelevartem/tether/issues/7) (Android, done); iOS, Desktop — _tbd_
+**GitHub Issues:** [#7](https://github.com/khmelevartem/tether/issues/7) (Android — done; iOS — done via Compose Multiplatform reuse, manually verified on simulator); Desktop — _tbd_ (entry point is CLI-only; `App()` not yet wired from desktop runner)
```
Comment (khmelevartem) [inline docs/product/features/device-list.md]: нужно кратко

File: docs/product/features/README.md [hunk 5]
```diff
@@ -35,7 +35,7 @@ Note on `Status` here vs. GitHub Issues: an issue can exist (and even be in prog
 |---------|------|--------|-----|--------|
 | mDNS peer discovery | Discovery | done | _tbd_ | #6 (iOS/macOS) |
 | Pairing | Pairing / Security | scoped | [pairing.md](pairing.md) | #9 (key exchange), #10 (PIN + CLI), #11 (Android UI); iOS, Desktop UI _tbd_ |
-| Device list screen | UI | in progress | [device-list.md](device-list.md) | #7 (Android); iOS, Desktop _tbd_ |
+| Device list screen | UI | in progress | [device-list.md](device-list.md) | #7 (Android + iOS via KMP UI reuse); Desktop _tbd_ |
```
Comment (khmelevartem) [inline docs/product/features/README.md]: нужно кратко

File: .claude/commands/work-on-issue.md (hunk 1 above)
Comment (khmelevartem) [reply .claude/commands/work-on-issue.md]: Принял — иначе новый блок smoke landed, но прогоняется только в следующем PR. Свёл Step 7 и Step 8 в один шаг с порядком a) расширь → b) прогоняй. Коммит e490ebf.

File: .claude/commands/work-on-issue.md (hunk 2 above)
Comment (khmelevartem) [reply .claude/commands/work-on-issue.md]: Re-audit на «не раздувай»: сжал все свои добавки в work-on-issue.md (Step 4 dep check, Step 7 smoke) — общий объём моих правок -40 строк. Заодно прошёлся тем же фильтром по retro.md (Step 2/3) и close-issue.md (Step 4.2/8) — они тоже компактнее. Коммит e490ebf.

File: docs/engineering/architecture-principles.md (hunk 3 above)
Comment (khmelevartem) [reply docs/engineering/architecture-principles.md]: Все три пункта применил: (1) поднял раздел над «Named classes over anonymous objects» — туда, где живут общие принципы, не decisions. (2) Переформулировал как «Common-first across KMP targets» — относится ко всему общему коду, не только UI; UI оставлен как самый видимый multiplier с #7 как референс. (3) Сжал. Коммит e490ebf.

File: docs/product/features/device-list.md (hunk 4 above)
Comment (khmelevartem) [reply docs/product/features/device-list.md]: Сократил до `#7 — Android + iOS done; Desktop tbd`. Коммит e490ebf.

File: docs/product/features/README.md (hunk 5 above)
Comment (khmelevartem) [reply docs/product/features/README.md]: Сократил до `#7 — Android + iOS done; Desktop _tbd_`. Коммит e490ebf.

File: .claude/commands/work-on-issue.md [hunk 6]
```diff
@@ -46,3 +53,15 @@
    - реализация им соответствует — или есть **осознанные отклонения** с обоснованием.
 
    Цель — поймать тихое расхождение между планом и кодом до того, как его поймает ревьюер. Если расхождение есть и не обосновано — почини до запроса review. Чек на уровне плана не заменяет этот: план и реализация легко расходятся по дороге.
+
+7. **runtime self-check (smoke).** До запроса ревью прогоняй релевантные блоки `/smoke-test` — задача исполнителя, не ревьюера. Ревьюер только проверяет наличие verdict'а в теле PR.
+
+   **a. Сначала — расширь smoke, если нужно.** PR ввёл новую критическую функциональность, которой нет в `/smoke-test`? Критическая = точка отказа на старте, happy-path фичи, cross-platform reusable UI, новый внешний интерфейс. Если да — допиши блок в `.claude/skills/smoke-test/SKILL.md` в этом же PR; крупное расширение — issue типа INFRA. **Не раздувай:** косметика и edge cases — в unit-тесты, не в smoke.
```
Comment (khmelevartem) [inline .claude/commands/work-on-issue.md]:
```suggestion
   **a. Сначала — расширь smoke, если нужно.** PR ввёл новую критическую функциональность, которой нет в `/smoke-test`? Критическая = точка отказа на старте, happy-path фичи, cross-platform reusable UI, новый внешний интерфейс. Если да — допиши блок в `.claude/skills/smoke-test/SKILL.md` в этом же PR. **Не раздувай:** помни, что смоук прогоняется часто, а тесты в нем дорогие по времени.
```

File: docs/engineering/architecture-principles.md [hunk 7]
```diff
@@ -28,6 +28,20 @@ Clean Architecture and adjacent patterns are full of ceremony that doesn't pay r
 - **Triple-layer DTO ↔ domain ↔ presentation mapping when there's no real divergence yet.** This one is nuanced: as the project matures, layers *do* legitimately need different shapes (DTO has serialization quirks, domain has invariants, presentation has display fields). The rule is not "never map" — it's "don't force three types up front." Start with one. Split when the second shape *actually* diverges (e.g. a UI list state with `selected`, `lastSeen`, `signal` is genuinely not the same as a network `Device`). Don't fight that growth in the name of brevity, and don't pre-empt it in the name of layering.
 - **Interfaces for things with one implementation.** An interface earns its place when there are at least two implementations, or it's a seam for testing. Otherwise the concrete class is the contract.
 
+## Common-first across KMP targets
+
+`CLAUDE.md` says: всё, что может жить в `commonMain` — там и лежит; платформенные source sets только для кода, требующего platform API. Это применимо ко всему — domain, network, presentation, UI. Реализация в `commonMain` ships на все активные таргеты одной кодовой базой.
```
Comment (khmelevartem) [inline docs/engineering/architecture-principles.md]:
```suggestion
Всё, что может жить в `commonMain` — там и лежит; платформенные source sets только для кода, требующего platform API. Это применимо ко всему — domain, network, presentation, UI. Реализация в `commonMain` поставляется на все активные таргеты одной кодовой базой.
```

File: docs/engineering/architecture-principles.md [hunk 8]
```diff
@@ -28,6 +28,20 @@ Clean Architecture and adjacent patterns are full of ceremony that doesn't pay r
...
+
+**Operational rules** (выводы из принципа, относятся ко всему common-коду — не только к UI):
+
+- Одна задача = одна общая реализация для всех shipping-платформ. Не создавай per-platform issues для функциональности, которая живёт в `commonMain`. Строки в `docs/product/features/README.md` отражают это.
+- `size:*` оценивается per-feature, не per-platform.
+- Компиляция ≠ корректность: код едет на все таргеты, но визуальная/runtime-корректность на каждом не гарантирована билдом. **Ручной smoke обязателен на каждой shipping-платформе** перед запросом ревью (см. `/work-on-issue` Step 7).
+
+**UI specifically** — самый видимый multiplier: Compose Multiplatform в `commonMain` ships на Android, iOS и Desktop (когда entry point вызывает `App()`) одной имплементацией. Подтверждено в #7 — экран написан под Android, на iOS заработал без модификаций.
+
```
Comment (khmelevartem) [inline docs/engineering/architecture-principles.md]:
```suggestion
- Компиляция ≠ корректность: код едет на все таргеты, но визуальная/runtime-корректность на каждом не гарантирована билдом. **Ручной smoke обязателен на каждой платформе** перед запросом ревью (см. `/work-on-issue` Step 7).

**UI specifically** — самый видимый multiplier: Compose Multiplatform в `commonMain` едет на Android, iOS и Desktop (когда entry point вызывает `App()`) одной имплементацией.

```

File: .claude/commands/work-on-issue.md [hunk 9]
```diff
@@ -46,3 +53,15 @@
...
+7. **runtime self-check (smoke).** До запроса ревью прогоняй релевантные блоки `/smoke-test` — задача исполнителя, не ревьюера. Ревьюер только проверяет наличие verdict'а в теле PR.
+
+   **a. Сначала — расширь smoke, если нужно.** PR ввёл новую критическую функциональность, которой нет в `/smoke-test`? Критическая = точка отказа на старте, happy-path фичи, cross-platform reusable UI, новый внешний интерфейс. Если да — допиши блок в `.claude/skills/smoke-test/SKILL.md` в этом же PR. **Не раздувай:** помни, что смоук прогоняется часто, а тесты в нем дорогие по времени.
+
+   **b. Затем прогоняй и записывай verdict.** Категории из `smoke-test/SKILL.md`: entry points → Android + iOS compile + Desktop CLI; FGS/manifest/Info.plist → Android; FileServer/CLI/mDNS → Desktop send; native source sets → native compile; `commonMain` UI → ручной smoke на **каждой** shipping-платформе (KMP-специфика: код едет на все таргеты, проверять надо все). Verdict в тело PR:
+   ```
+   ## Smoke verdict
+   - Block 3 (Android FGS): 🟢
+   - Manual iOS UI (KMP reuse): 🟢
+   ```
```
Comment (khmelevartem) [inline .claude/commands/work-on-issue.md]:
```suggestion
   **b. Затем прогоняй релевантные блоки и записывай verdict** в тело PR:
   ```
   ## Smoke verdict
   - Block 3 (Android FGS): 🟢
   - Manual iOS UI (KMP reuse): 🟢
   ```
```
